Documentation/Viper RxModule Guide.md

```diff
@@ -0,0 +1,560 @@
+            }
+            .asDriver(onErrorDriveWith: .never())
+            .startWith(false)
+    }
+
+    func initializeLogin(with login: Signal<Void>, email: Driver<String?>, password: Driver<String?>) {
+
+        let inputs = Driver.combineLatest(
+            email.asObservable().compactMap { $0 }.asDriver(onErrorDriveWith: .empty()),
+            password.asObservable().compactMap { $0 }.asDriver(onErrorDriveWith: .empty())
+        )
+
+        login
+            .withLatestFrom(inputs)
+            .flatMap { [unowned self] email, password -> Signal<User> in
+                interactor.login(with: email, password)
+                    .do(onError: { [unowned self] error in
+                        view.hideProgressHUD()
+                        wireframe.showErrorAlert(with: error.localizedDescription)
+                    }, onSubscribe: { [unowned view] in
+                        view.showProgressHUD()
+                    })
+                    .asSignal(onErrorSignalWith: .never())
+            }
+            .do(onNext: { [unowned self] user in
+                authorizationManager.authorizationHeader = user.authorizationHeader
+            })
+            .emit(onNext: { [unowned wireframe] _ in
+                wireframe.navigate(to: .home)
+            })
+            .disposed(by: disposeBag)
+    }
+
+    func isEmailValid(_ email: String) -> Bool {
+        return emailValidator.isValid(email)
+    }
+
+    func isPasswordValid(_ password: String) -> Bool {
+        return passwordValidator.isValid(password)
+    }
+
+    func initializeRegister(with register: Signal<Void>) {
+        register
+            .emit(onNext: { [unowned wireframe] in
+                wireframe.navigate(to: .register)
+            })
+            .disposed(by: disposeBag)
+    }
+}
+```
+
+Whoa! That was a lot of code. Don't fret, it's quite simple :)
+
+We'll start from the top. As the base example, we have implemented some part of validation for our email and password. Now we're going straight down to the <i>configure</i> function. In the <i>configure</i> function we have registered to the events from the *ViewOutput*. As there is a driver which we'll send to the view, we have to call <i>initializeInputs</i> function and pass it as a parameter. We'll take both, email and password drivers, do some Rx magic which will check if the inputs are valid, and return a fresh driver with a bool value. If the inputs are valid it returns true, and as the name of the driver says, button will be available, otherwise, it's disabled.
+
+After we have subscribed to our inputs, we want to subscribe to button taps. Since they only want to navigate or request an API call, we don't have to return anything from their initialization. That's why we sometimes need the disposeBag in the Presenter. That is mostly about the Presenter, we use a lot of Rx a try to stay as pure with Rx as possible.
+
+Next, we'll get in touch with our implemented *RxLoginViewController*.
```

Comment:
Not sure about this chunk, it wasn't clear to me while I was reading, but I guess you meant this? 

```suggestion
We'll start from the top. At the base example, we have implemented some part of validation for the email and password. Now we're going straight down to the <i>configure</i> function. Inside the <i>configure</i> method presenter registers to the events from the *ViewOutput*. As there is a driver which we'll send to the view, we have to call <i>initializeInputs</i> function and pass it as a parameter. We'll take both, email and password drivers, do some Rx magic which will check if the inputs are valid, and return a fresh driver with a bool value. If the inputs are valid it returns true, and as the name of the driver says, button will be available, otherwise, it's disabled.

After presenter makes inputs for the view, we want the presenter to subscribe to the actions that are performed by the view. Since those actions only result in navigation or API call, we don't have to return anything, just handle the navigation/calls internally. 

Next, we'll get in touch with our implemented *RxLoginViewController*.
```

Reply:
Btw, try to define more explicit wording when writing, it's not we who are subscribing, it's presenter/view.